test/parsers/JsonUnitLinkEditBodyParserSpec.scala

```diff
@@ -73,6 +88,15 @@ class JsonUnitLinkEditBodyParserSpec extends FreeSpec with Matchers with GuiceOn
           result.left.value.header.status shouldBe BAD_REQUEST
         }
       }
+
+      "when the patch document valid json but does not comply with our model (before/after not from/to)" in new Fixture {
+        val request = FakeRequest().withHeaders(CONTENT_TYPE -> JSON)
+        val body = Source.single(ByteString(UnprocessableVATEditParentLinkPostBody))
+
+        whenReady(JsonUnitLinkEditBodyParser.apply(request).run(body)) { result =>
+          result.left.value.header.status shouldBe UNPROCESSABLE_ENTITY
+        }
+      }
     }
   }
```

Comment:
Fine as is stands - but this is pretty broad brush.
What if I have a "from" but it does not contain an "id"?
What if the "id" is a JsNumber rather than a JsString?
What if "type" does not contain a valid unit type?

Reply:
I'll add in those other test cases!